addons/payment_test/controllers/main.py

```diff
@@ -1,26 +1,57 @@
 # -*- coding: utf-8 -*-
 # Part of Odoo. See LICENSE file for full copyright and licensing details.
 
-from odoo import http, _
+import logging
+from random import randint
+
+from odoo import http
 from odoo.http import request
-from odoo.exceptions import UserError
+
+from odoo.addons.payment import utils as payment_utils
+
+_logger = logging.getLogger(__name__)
 
 
 class PaymentTestController(http.Controller):
 
-    @http.route(['/payment/test/s2s/create_json_3ds'], type='json', auth='public', csrf=False)
-    def payment_test_s2s_create_json_3ds(self, verify_validity=False, **kwargs):
-        if not kwargs.get('partner_id'):
-            kwargs = dict(kwargs, partner_id=request.env.user.partner_id.id)
-        acquirer = request.env['payment.acquirer'].browse(int(kwargs.get('acquirer_id')))
-        if acquirer.state != 'test':
-            raise UserError(_("Please do not use this acquirer for a production environment!"))
-        token = acquirer.s2s_process(kwargs)
-
-        return {
-            'result': True,
-            'id': token.id,
-            'short_name': 'short_name',
+    @http.route('/payment/test/payment', type='json', auth='public')
+    def process_payment(self, reference, cc_number, cc_cvc, cc_name, cc_expiry):
+        """ Simulate the result of a payment request and handle the response.
+
+        :param str reference: The reference of the transaction
+        :param int cc_number: The credit card number
+        :param int cc_cvc: The credit card cvc
+        :param int cc_name: the credit card owner
+        :param int cc_expiry: The credit card expiry date
+        :return: The JSON-formatted content of the response
+        :rtype: dict
+        """
+        fake_api_response = {
+            'status': 'success',
             '3d_secure': False,
             'verified': True,
+            'acquirer_reference': randint(1, 10000),
+            'cc_number': payment_utils.build_token_name(payment_details_short=cc_number[-4:]),
```

Comment:
```suggestion
            'cc_summary': cc_number[-4:],
```
Let's not store two different things under the same name.
`payment_utils.build_token_name` can later be called on `cc_summary` when actually creating a token.